tests/Unit/DataStructures/Tensor/EagerMath/Test_Norms.cpp

```diff
@@ -0,0 +1,204 @@
+                        InverseMetric<1, Frame>, InverseMetric<2, Frame>,
+                        InverseMetric<3, Frame>>,
+      db::AddComputeTags<Tags::PointwiseL2NormCompute<MyScalar>,
+                         Tags::PointwiseL2NormCompute<Vector<1, Frame>>,
+                         Tags::PointwiseL2NormCompute<Vector<2, Frame>>,
+                         Tags::PointwiseL2NormCompute<Vector<3, Frame>>,
+                         Tags::PointwiseL2NormCompute<Covector<1, Frame>>,
+                         Tags::PointwiseL2NormCompute<Covector<2, Frame>>,
+                         Tags::PointwiseL2NormCompute<Covector<3, Frame>>,
+                         Tags::PointwiseL2NormCompute<Metric<1, Frame>>,
+                         Tags::PointwiseL2NormCompute<Metric<2, Frame>>,
+                         Tags::PointwiseL2NormCompute<Metric<3, Frame>>,
+                         Tags::PointwiseL2NormCompute<InverseMetric<1, Frame>>,
+                         Tags::PointwiseL2NormCompute<InverseMetric<2, Frame>>,
+                         Tags::PointwiseL2NormCompute<InverseMetric<3, Frame>>,
+                         Tags::L2NormCompute<MyScalar>,
+                         Tags::L2NormCompute<Vector<1, Frame>>,
+                         Tags::L2NormCompute<Vector<2, Frame>>,
+                         Tags::L2NormCompute<Vector<3, Frame>>,
+                         Tags::L2NormCompute<Covector<1, Frame>>,
+                         Tags::L2NormCompute<Covector<2, Frame>>,
+                         Tags::L2NormCompute<Covector<3, Frame>>,
+                         Tags::L2NormCompute<Metric<1, Frame>>,
+                         Tags::L2NormCompute<Metric<2, Frame>>,
+                         Tags::L2NormCompute<Metric<3, Frame>>,
+                         Tags::L2NormCompute<InverseMetric<1, Frame>>,
+                         Tags::L2NormCompute<InverseMetric<2, Frame>>,
+                         Tags::L2NormCompute<InverseMetric<3, Frame>>>>(
+      db::item_type<MyScalar>{{minus_three}},
+      db::item_type<Vector<1, Frame>>{{{minus_three}}},
+      db::item_type<Vector<2, Frame>>{{{minus_three, minus_three}}},
+      db::item_type<Vector<3, Frame>>{{{minus_three, minus_five, four}}},
+      db::item_type<Covector<1, Frame>>{{{four}}},
+      db::item_type<Covector<2, Frame>>{{{four, two}}},
+      db::item_type<Covector<3, Frame>>{{{four, two, twelve}}}, psi_1d, psi_2d,
+      psi_3d, determinant_and_inverse(psi_1d).second,
+      determinant_and_inverse(psi_2d).second,
+      determinant_and_inverse(psi_3d).second);
+
+  // Test point-wise L2-norm against precomputed values
+  // rank 0
+  CHECK(get(db::get<Tags::PointwiseL2Norm<MyScalar>>(box)) ==
+        DataVector{npts, 3.0});
+  // rank (1, 0)
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Vector<1, Frame>>>(box)) ==
+        DataVector{npts, 3.});
+  CHECK(get(db::get<Tags::PointwiseL2Norm<Vector<2, Frame>>>(box)) ==
+        DataVector{npts, 4.242640687119285});
```

Comment:
These should probably be `CHECK_ITERABLE_APPROX` to avoid floating point precision errors.

Reply:
changed